### PR TITLE
Add filter to allow autoshare to be enabled by default for a post type

### LINKED ISSUE #68: Add filter to turn on autoshare by default on a per-post-type basis
**Is your enhancement related to a problem? Please describe.**

Autoshare is off by default with no way of turning it on by default. This enhancement will provide a way to opt post types in by default at the code level.

**Describe the solution you'd like**

A simple WP filter that can be hooked into by themes/other plugins. The value will be passed to JS and will set the default in the editor.

**Designs**

**Describe alternatives you've considered**

There should be a UI feature associated with this, but that can be part of #57.

**Additional context**

### PR DESCRIPTION
**Pull request by johnwatkins0**
### Description of the Change

This adds a filter, `autoshare_for_twitter_enabled_default`, that receives the post type of the current post and allows the default enabled value for the post type to be overridden. Without the filter applied, the default is false.

### Alternate Designs

#57 should introduce a corresponding UI component for this -- e.g., a series of checkboxes for public post types.

### Benefits

Less work for editors when they know a post type will always be shared.

### Possible Drawbacks

Using the update requires a small amount of engineering effort to apply the filter.

### Verification Process

Add this filter to a theme or plugin on the same site:

```php
function enable_autoshare_by_default_for_core_post_type( $enabled, $post_type ) {
	if ( 'post' === $post_type ) {
		return true;
	}

	return $enabled;
}
add_filter( 'autoshare_for_twitter_enabled_default', 'enable_autoshare_by_default_for_core_post_type', 10, 2 );
```

(This example is also included in this PR in README.md.)

When this filter is active, autoshare should be enabled by default for the core `post` post type but not on any other post types. When the filter is removed, autoshare should again be disabled by default.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Applicable Issues

Resolves #68

### Changelog Entry

Add filter to allow autoshare to be enabled by default for a post type.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,20 @@ add_action( 'init', 'opt_my_cpt_into_autoshare' );
...
+
+```php
+function enable_autoshare_by_default_for_core_post_type( $enabled, $post_type ) {
+	if ( 'post' === $post_type ) {
+		return true;
+	}
+
+	return $enabled;
+}
+add_filter( 'autoshare_for_twitter_enabled_default' 'enable_autoshare_by_default_for_core_post_type', 10, 2 );
```
Comment (rickalee) [inline README.md]: @johnwatkins0 Missing comma between filter and function name.

Comment (johnwatkins0) [reply README.md]: Thanks! Updated.

**Comment by rickalee** [comment]
@johnwatkins0 Should we be showing checkbox as checked in Classic Editor once filter is in place?

https://rickal.ee/p9u5mk90

**Comment by johnwatkins0** [comment]
Good call, @rickalee . Added this commit to pass the enabled status to the classic editor checkbox: https://github.com/10up/autoshare-for-twitter/pull/71/commits/60af3b7792ce61b720efe7d517ce37755fa63f57

**Comment by rickalee** [comment]
@johnwatkins0 Checkbox is now checked but on Publish it doesn't tweet.

https://rickal.ee/DOuv7mep

**Comment by johnwatkins0** [comment]
Thanks, @rickalee This last commit should do it: https://github.com/10up/autoshare-for-twitter/pull/71/commits/13bbad40627591d58a35ec0948e26a7ea8481f30 Did an extra round off testing myself this time 😅

File: includes/utils.php [hunk 2]
```diff
@@ -68,7 +100,7 @@ function delete_autoshare_for_twitter_meta( $id, $key ) {
  * @return bool
  */
 function maybe_autoshare( $post_id ) {
-	return ( 1 === intval( get_autoshare_for_twitter_meta( $post_id, ENABLE_AUTOSHARE_FOR_TWITTER_KEY ) ) ) ? true : false;
+	return ( 1 === intval( autoshare_enabled( $post_id ) ) ) ? true : false;
```
Comment (dinhtungdu) [inline includes/utils.php]: Should we remove this function? `maybe_autoshare()` is only used in `post-transition.php:88`. And it's just a wrapper for `autoshare_enabled`.

Comment (johnwatkins0) [reply includes/utils.php]: Yes, good idea. I have updated in https://github.com/10up/autoshare-for-twitter/pull/71/commits/bf7f4444d9e8366977c64179a9052d72cfaa86fa

**Pull request merged**